lib/hashie/extensions/hash_initializer.rb

```diff
@@ -0,0 +1,24 @@
+module Hashie
+  module Extensions
+    # Marker module to indicate has a single-argument hash constructor.
+    module HashInitializer
```

Comment:
Note: I created `Hashie::Extensions::HashInitializer` that's just a marker module, because I didn't have a way to tell what classes could be initialized from a Hash. That's the behavior for `Mash` and `Dash`, but that's not how the initializer works (by default) for `::Hash` or `Hashie::Hash`. It *is* the behavior if you mixin `Hashie::Extensions::MergeInitializer`, though. I'm open to other ideas, though, perhaps a `.from_hash` method instead of a marker module?